src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -815,11 +820,17 @@ public async Task<IServiceChannelDispatcher> CreateServiceChannelDispatcherAsync
                             {
                                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new CommunicationException(SR.Format(SR.SecuritySessionNotPending, _sessionToken.ContextId)));
                             }
-                            if (_settings.AcceptorChannelType is IDuplexChannel)
+                            if (_settings.AcceptorChannelType == typeof(IDuplexSessionChannel))
                             {
-                                throw new PlatformNotSupportedException();
+                                ServerSecurityDuplexSessionChannel.ServerSecurityDuplexSessionChannelDispatcher
+                                 _replySessionDuplexChannelDispatcher = new ServerSecurityDuplexSessionChannel.
+                                 ServerSecurityDuplexSessionChannelDispatcher(_settings, _sessionToken,
+                                 null, _settings.SettingsLifetimeManager, channel, _remoteAddress);
+                                await _replySessionDuplexChannelDispatcher.OpenAsync(ServiceDefaults.OpenTimeout);
+                                _sessionChannelDispatcher = _replySessionDuplexChannelDispatcher;
+                                _settings.AddSessionChannel(_sessionToken.ContextId, _replySessionDuplexChannelDispatcher, _messageFilter);
```

Comment:
Can some of the duplicated code here be refactored out and placed after the if/else if block?

Reply:
done